### DISCUSSION TITLE
How do I add product filters to the frontend?

### DISCUSSION THREAD

**Discussion by aashan10** [coreshop/CoreShop#2656] [Q&A]
![Screenshot 2024-06-25 at 14 00 57](https://github.com/coreshop/CoreShop/assets/18713900/9475820e-f643-4f34-a0e2-c062b1c4912a)

I'm trying out coreshop for a potential customer and I need filters in product listing page. I need a product filter in category pages like the one shown in the picture above. How can I do that? I searched on the documentation but could not find the right guide for this. Is this available with vanilla coreshop instance or are there any other bundles that needs to be installed?

**Comment by aashan10** [comment]
@dpfaffenbauer if you could point me to the right direction, that'd be awesome.

**Reply by dpfaffenbauer** [reply]
yes, this is under Product -> Index for Indexing, and Product -> Filter for the Frontend Configuration. You can then configure the filter on category level and configure it tohave it similar to your screenshot. You also need to implement some templates for the color and index the price for the min-max filter

**Reply by twin-elements** [reply]
If we have a user logged in who has discounts assigned using **price rules**.
Are the prices from price rules also indexed and can max-min price filters be built from them?
![image](https://github.com/user-attachments/assets/31f16e68-e430-46c8-8b83-b27b3f9d06a1)
![image](https://github.com/user-attachments/assets/c8e76f7b-702c-4f36-bd23-89154afb4b39)

**Reply by dpfaffenbauer** [reply]
@twin-elements unfortunately not...

**Reply by twin-elements** [reply]
@dpfaffenbauer And would you have any idea how such a thing could be done in Coreshop? Of course I can help, but so far I am not 100% familiar with the functionality of Coreshop.

**Reply by dpfaffenbauer** [reply]
Depends on the complexity of your rules. There are so many different factors playing into the price calculation. If you have a very simple way, you could create separate index fields for the prices and filter on these. There is no general solution to this...
